### PR TITLE
bespokesynth: init at 1.0.0

### PR DESCRIPTION
**Pull request by astro**
###### Motivation for this change

> Software modular synth with controllers support, scripting and VST

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/bespokesynth/default.nix [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    sha256 = "04b2m40jszphslkd4850jcb8qwls392lwy3lc6vlj01h4izvapqk";
+  };
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+    ${xvfb-run}/bin/xvfb-run sh -e <<EOF
+      ${projucer}/bin/Projucer --set-global-search-path linux defaultJuceModulePath ${projucer.src}/modules
+      ${projucer}/bin/Projucer --resave BespokeSynth.jucer
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/bespokesynth/default.nix:64]:
```suggestion
    xvfb-run sh -e <<EOF
      ${projucer}/bin/Projucer --set-global-search-path linux defaultJuceModulePath ${projucer.src}/modules
      ${projucer}/bin/Projucer --resave BespokeSynth.jucer
```
Please add xvfb-run to nativeBuildInputs.

File: pkgs/applications/audio/bespokesynth/default.nix [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  };
+
+  # equal to vst-sdk in ../oxefmsynth/default.nix
+  vst-sdk = stdenv.mkDerivation rec {
+    name = "vstsdk3610_11_06_2018_build_37";
+    src = fetchzip {
+      url = "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/${name}.zip";
+      sha256 = "0da16iwac590wphz2sm5afrfj42jrsnkr1bxcy93lj7a369ildkj";
+    };
+    installPhase = "cp -r . $out";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/bespokesynth/default.nix:43]:
```suggestion
    installPhase = ''
      cp -r . $out
    '';
```

File: pkgs/applications/audio/bespokesynth/default.nix [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      rev = version;
+      sha256= "0qpiqfwwpcghk7ij6w4vy9ywr3ryg7ppg77bmd7783kxg6zbhj8h";
+    };
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [
+      freetype libX11 libXrandr libXinerama libXext gtk3 webkitgtk
+      libjack2 curl
+    ];
+    preBuild = "cd extras/Projucer/Builds/LinuxMakefile";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/bespokesynth/default.nix:26]:
```suggestion
    preBuild = ''
      cd extras/Projucer/Builds/LinuxMakefile
    '';
```

File: pkgs/applications/audio/bespokesynth/default.nix [hunk 4]
```diff
@@ -0,0 +1,109 @@
+{ lib, stdenv, fetchFromGitHub, pkg-config, fetchzip
+, libjack2, alsa-lib, freetype, libX11, libXrandr, libXinerama, libXext, libXcursor
+, libGL, python3, ncurses, libusb
+, gtk3, webkitgtk, curl, xvfb-run, makeWrapper
+  # "Debug", or "Release"
+, buildType ? "Release"
+}:
+
+let
+  projucer = stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/bespokesynth/default.nix:10]: Maybe we should move projucer to top-level.

Comment (astro) [reply pkgs/applications/audio/bespokesynth/default.nix:10]: Now that they switched to cmake after the release, we will be able to remove projucer for this package with the next update.

**Pull request merged**